### PR TITLE
Feature/groups support

### PR DESCRIPTION
**Pull request by atrejut**

### CODE REVIEW

File: sailor/assetcentral/group.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+        member_name = set_class._element_type.__name__
+
+        if kwargs.get('id'):
+            raise RuntimeError(f'Can not specify `id` when retrieving "{member_name}" from a group.')
+
+        kwargs['id'] = [item['businessObjectId'] for item in self._members_raw
+                        if item['businessObjectType'] == business_object_type]
+
+        if not kwargs['id']:
+            warnings.warn(f'There are no "{member_name}" in this group!')
```
Comment (cloudyday) [inline sailor/assetcentral/group.py:49]: You could use the DataNotFoundWarning we have for cases like this. Might not be completely accurate though. Give it a thought.

File: sailor/assetcentral/group.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+    def _generic_get_members(self, business_object_type, set_class, find_function, extended_filters, **kwargs):
+        member_name = set_class._element_type.__name__
+
+        if kwargs.get('id'):
+            raise RuntimeError(f'Can not specify `id` when retrieving "{member_name}" from a group.')
+
+        kwargs['id'] = set([item['businessObjectId'] for group in self.elements for item in group._members_raw
+                            if item['businessObjectType'] == business_object_type])
+        if not kwargs['id']:
+            warnings.warn(f'There are no "{member_name}" in any of the groups in this set!')
```
Comment (cloudyday) [inline sailor/assetcentral/group.py:125]: You could use the DataNotFoundWarning we have for cases like this. Might not be completely accurate though. Give it a thought.

File: sailor/assetcentral/group.py [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+
+    Find all Groups which either have the name 'MyGroup' or the name 'MyOtherGroup'::
+
+        find_group(name=['MyGroup', 'MyOtherGroup'])
+
+    Find all Groups with the name 'MyGroup' which are also of type 'FLEET'::
+
+        find_groups(name='MyGroup', group_type='FLEET')
+
+    Find all Groups  having a risk_value greater than 0::
```
Comment (cloudyday) [inline sailor/assetcentral/group.py:210]: one space too many

File: sailor/assetcentral/group.py [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+Group module can be used to retrieve Groups information from AssetCentral.
+
+Classes are provided for individual Groups as well as groups of Groups (GroupSet).
+"""
+
+from functools import cached_property
+import warnings
+
+from .utils import AssetcentralEntity, ResultSet, _ac_application_url, _fetch_data, _apply_filters_post_request, \
+    _add_properties
```
Comment (cloudyday) [inline sailor/assetcentral/group.py:11]: for new code we should use the recommended way of doing multi-line imports using parentheses (tuple). I also started using this when I had to change one of these import lines in my former PRs.
https://www.python.org/dev/peps/pep-0328/#rationale-for-parentheses

**Review by cloudyday — COMMENTED** [review]
looks really great

File: sailor/assetcentral/group.py (hunk 2 above)
Comment (atrejut) [reply sailor/assetcentral/group.py:125]: I think I'd rather keep it for cases where the data should be coming from a remote system?

**Review by cloudyday — APPROVED** [review]
LGTM

**Pull request merged**